src/Service/ServiceExe.cs

```
@@ -1791,42 +1791,58 @@ public async Task<bool> TryVerifyContainerConnection()
-                {
-                    // Verify it is still working properly
-                    await _containerClient.GetServiceVersionString();
-                }
-                else
-                {
-                    // Initial connection made or reconnected
-                    LogServiceEvent(new ServiceEvent(ServiceEventType.ContainerConnected, ContainerGuid, Resources.ContainerConnected));
+                    if (previousContainerStatus == IsContainerConnected)
+                    {
+                        // Verify it is still working properly
+                        await _containerClient.GetServiceVersionString();
+                    }
+                    else
+                    {
+                        // Initial connection made or reconnected
+                        LogServiceEvent(new ServiceEvent(ServiceEventType.ContainerConnected, ContainerGuid, Resources.ContainerConnected));
+                    }
+
+                    // Connection established, exit loop
+                    break;
                 }
-            }
-            catch (FactoryOrchestratorContainerException)
-            {
-                if (previousContainerStatus == IsContainerConnected)
+                catch (FactoryOrchestratorContainerException)
                 {
-                    // Connection lost
-                    LogServiceEvent(new ServiceEvent(ServiceEventType.ContainerDisconnected, previousContainerGuid, Resources.ContainerDisconnected));
-                }
+                    if (previousContainerStatus == IsContainerConnected)
+                    {
+                        // Connection lost
+                        LogServiceEvent(new ServiceEvent(ServiceEventType.ContainerDisconnected, previousContainerGuid, Resources.ContainerDisconnected));
+                    }
 
-                throw;
+                    if (w.ElapsedMilliseconds * 1000 > retrySeconds)
+                    {
+                        throw;

```

would we want to throw a timeout exception here?

---

No, I only want container APIs to throw exceptions of type FactoryOrchestratorContainerException. I think the fact that it timed out is more of an implementation detail that a consumer doesn't need to know.

---

Ah gotcha, makes sense